samples/extensions/subgroups_operations/subgroups_operations.cpp

```diff
@@ -0,0 +1,503 @@
+}
+
+void SubgroupsOperations::setup_descriptor_pool()
+{
+	std::vector<VkDescriptorPoolSize> pool_sizes = {
+	    vkb::initializers::descriptor_pool_size(VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1u),
+	    vkb::initializers::descriptor_pool_size(VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER, 1u)};
+	VkDescriptorPoolCreateInfo descriptor_pool_create_info =
+	    vkb::initializers::descriptor_pool_create_info(
+	        static_cast<uint32_t>(pool_sizes.size()),
+	        pool_sizes.data(),
+	        2u);
+	VK_CHECK(vkCreateDescriptorPool(get_device().get_handle(), &descriptor_pool_create_info, nullptr, &descriptor_pool));
+}
+
+void SubgroupsOperations::setup_descriptor_set_layout()
+{
+	std::vector<VkDescriptorSetLayoutBinding> set_layout_bindings =
+	    {
+	        vkb::initializers::descriptor_set_layout_binding(
+	            VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,
+	            VK_SHADER_STAGE_VERTEX_BIT,
+	            0u),
+	        vkb::initializers::descriptor_set_layout_binding(
+	            VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
+	            VK_SHADER_STAGE_FRAGMENT_BIT,
+	            1u)};
+	VkDescriptorSetLayoutCreateInfo descriptor_layout =
+	    vkb::initializers::descriptor_set_layout_create_info(
+	        set_layout_bindings.data(),
+	        static_cast<uint32_t>(set_layout_bindings.size()));
+	VK_CHECK(vkCreateDescriptorSetLayout(get_device().get_handle(), &descriptor_layout, nullptr, &texture_object.descriptor_set_layout));
+	VkPipelineLayoutCreateInfo pipeline_layout_create_info =
+	    vkb::initializers::pipeline_layout_create_info(
+	        &texture_object.descriptor_set_layout,
+	        1u);
+	VK_CHECK(vkCreatePipelineLayout(get_device().get_handle(), &pipeline_layout_create_info, nullptr, &texture_object.pipeline.pipeline_layout));
+}
+
+void SubgroupsOperations::setup_descriptor_set()
+{
+	VkDescriptorSetAllocateInfo alloc_info =
+	    vkb::initializers::descriptor_set_allocate_info(
+	        descriptor_pool,
+	        &texture_object.descriptor_set_layout,
+	        1u);
+	VK_CHECK(vkAllocateDescriptorSets(get_device().get_handle(), &alloc_info, &texture_object.descriptor_set));
+	VkDescriptorBufferInfo            buffer_descriptor = create_descriptor(*texture_uniform_buffer);
+	VkDescriptorImageInfo             image_descriptor  = create_descriptor(texture_object.texture);
+	std::vector<VkWriteDescriptorSet> write_descriptor_sets =
+	    {
+	        vkb::initializers::write_descriptor_set(
+	            texture_object.descriptor_set,
+	            VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER,
+	            0u,
+	            &buffer_descriptor),
+	        vkb::initializers::write_descriptor_set(
+	            texture_object.descriptor_set,
+	            VK_DESCRIPTOR_TYPE_COMBINED_IMAGE_SAMPLER,
+	            1u,
+	            &image_descriptor)};
+	vkUpdateDescriptorSets(get_device().get_handle(), static_cast<uint32_t>(write_descriptor_sets.size()), write_descriptor_sets.data(), 0u, NULL);
```

Comment:
How about using nullptr instead of NULL? The last argument of vkUpdateDescriptorSets is a pointer to an array of VkCopyDescriptorSet so nullptr better suggest that is the pointer

Reply:
out to date